### ISSUE TITLE
gdal_utils(util = "mdiminfo") fails on file for which command-line equivalent succeeds

### ISSUE THREAD

**Issue by JoshOBrien** [r-spatial/sf#2020]
**Describe the bug**

Running `gdal_utils()` with `"gdaldiminfo"` fails where it seems it should succeed.

**To Reproduce**

To set up a reproducible example, based on [this example from SO](https://stackoverflow.com/a/28984566/980833), I first create an example NetCDF file:

```r
library(ncdf4)

## Define the netCDF variable
xvals <- seq(-177.5, 177.5, 5)
yvals <- seq(-87.5, 87.5, 5)
nx <- length(xvals)
ny <- length(yvals)
lon1 <- ncdim_def("longitude", "degrees_east", xvals)
lat2 <- ncdim_def("latitude", "degrees_north", yvals)
time <- ncdim_def("Time", "months", 1:12, unlim = TRUE)
mv <- -999 ## missing value to use
var_temp <-
    ncvar_def(name = "temperature",
              units = "Celsius",
              dim = list(lon1, lat2, time),
              missval = -999,
              longname = "Dummy_SST_Dataset")

## Create NetCDF file
filename <- "time.nc"
ncnew <- nc_create(filename, list(var_temp))
data <- rep(yvals, each = nx)
data[sample(1:(nx*ny), 100, replace = FALSE)] <- -999
ncvar_put(ncnew, var_temp, data, start = c(1, 1, 1), count = c(nx, ny, 1))
nc_close(ncnew)
```

From the command line, on a Windows box with a local installation of GDAL, this works just fine:

```
gdalmdiminfo time.nc
```
but running the **sf** package equivalent fails like so:

```r
library(sf)
gdal_utils("mdiminfo", source = "time.nc")
## Error in CPL_gdalmdiminfo(source, options, oo) : 
##   GDALMultiDimInfo returned NULL
```

I'm not sure whether there is some problem with the way I'm calling `gdal_utils()`, or if there is an issue with `gdal_utils()` itself.

Am happy to provide any additional info if that would be helpful.

**Comment by edzer** [comment]
Should be fixed in 1.0-9, did you try that, e.g. from r-universe? See https://github.com/r-spatial/sf/blob/main/NEWS.md

**Comment by JoshOBrien** [comment]
My bad. As you say, it's fixed in 1.0-9. Thanks.
